web/src/main/java/org/mskcc/cbio/oncokb/controller/CacheController.java

```diff
@@ -57,42 +62,51 @@ Map<String, Object> getGeneCache(
         return result;
     }
 
-    @RequestMapping(value = "/legacy-api/cache", method = RequestMethod.POST, produces = "application/json")
-    public
+    @PostMapping(value = "/legacy-api/cache", produces = "application/json")
     @ResponseBody
-    Map<String, String> postAlteration(
-        HttpMethod method,
-        @RequestParam(value = "cmd", required = false) String cmd,
+    public Map<String, String> handleCacheCommand(
+        @RequestParam(value = "cmd", required = true) String cmd,
         @RequestParam(value = "entrezGeneIds", required = false) Set<Integer> entrezGeneIds,
         @RequestParam(value = "propagation", required = false, defaultValue = "false") Boolean propagation
-    ) throws IOException {
-        Map<String, String> result = new HashMap<>();
-        if (cmd != null) {
+    ) {
+        Map<String, String> result = new LinkedHashMap<>();
+    
+        if (cmd == null || cmd.isEmpty()) {
+            result.put("status", "error");
+            result.put("message", "Missing 'cmd' parameter");
+            return result;
+        }
+    
+        try {
             switch (cmd) {
                 case "reset":
-                    resetCache(propagation);
+                    CacheUtils.resetAll(propagation);
+                    result.put("status", "success");
+                    result.put("message", "All in memory caches reset");
                     break;
+    
                 case "updateGene":
+                    if (entrezGeneIds == null || entrezGeneIds.isEmpty()) {
+                        result.put("status", "error");
+                        result.put("message", "Missing or empty 'entrezGeneIds' for updateGene");
+                        return result;
+                    }
                     CacheUtils.updateGene(entrezGeneIds, propagation);
+                    result.put("status", "success");
+                    result.put("message", "Updated in memory cache");
                     break;
-                case "updateAbbreviationOntology":
-                    NamingUtils.cacheAllAbbreviations();
-                    break;
+    
                 default:
+                    result.put("status", "error");
+                    result.put("message", "Invalid command: " + cmd);
                     break;
             }
-        }
-        result.put("status", "success");
-        return result;
-    }
-
-    private Boolean resetCache(Boolean propagation) {
-        Boolean operation = true;
-        try {
-            CacheUtils.resetAll(propagation);
         } catch (Exception e) {
-            operation = false;
+            LOGGER.error("Cache operation failed for cmd={} (propagation={})", cmd, propagation);
```

Comment:
Include the exception in the log like so:
```java
LOGGER.error("Cache operation failed for cmd={} (propagation={})", cmd, propagation, e);
```

This allows you configure out exception show up via the log4j configuration.